sources_test.go

```diff
@@ -254,3 +254,49 @@ func TestStdin(t *testing.T) {
 		t.Errorf("want %q, got %q", want, string(got))
 	}
 }
+
+func TestPrompt(t *testing.T) {
+	tcs := []struct {
+		Name        string
+		Input       string
+		Default     string
+		Want        string
+		ErrExpected bool
+	}{
+		{"User Input OK", "123", "notused", "123", false},
+		{"No User Input", "", "default", "default", false},
+		{"Complex Input", "/user/test @@@ testing one two three", "default", "/user/test @@@ testing one two three", false},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.Name, func(t *testing.T) {
+			want := []byte(tc.Input)
+
+			// Prepare Stdin before calling prompt.
+			tmp, err := ioutil.TempFile("", tc.Name)
+			if err != nil {
+				t.Error(err)
+			}
+			defer os.Remove(tmp.Name())
+			if _, err := tmp.Write(want); err != nil {
+				t.Error(err)
+			}
+
+			if _, err := tmp.Seek(0, 0); err != nil {
+				t.Error(err)
+			}
+
+			// Restore original Stdin.
+			oldStdin := os.Stdin
+			defer func() { os.Stdin = oldStdin }()
+			os.Stdin = tmp
```

Comment:
Not concurrency-safe; see `TestStdin` for a safe way to do this.

Reply:
I am having a little trouble on the implementation of this test.

When utilizing the ```exec.Command``` method, with a new **SCRIPT_TEST** defined for ```prompt``` as 
```go
case "prompt":
		Prompt("Input", "Default").Stdout()
```

The ```cmd.Output()``` is actually the first Stdout message from ```Prompt```, which is the message to the user ```Echo(fmt.Sprintf("%s [%s]: ", message, defaultValue)).Stdout()```